src/python/pants/backend/jvm/tasks/ivy_task_mixin.py

```diff
@@ -101,6 +102,8 @@ def implementation_version(cls):
     return super(IvyTaskMixin, cls).implementation_version() + [('IvyTaskMixin', 4)]
 
   @memoized_property
+  @deprecated(removal_version='1.10.0.dev0',
+              hint_message='Use ivy_repository_cache_dir or ivy_resolution_cache_dir instead.')
   def ivy_cache_dir(self):
```

Comment:
@cheister Can you clarify when `ivy_cache_dir` will be used instead of the two new options?

Reply:
I think in most cases people would use the ivy_cache_dir, since most of the time you want the resolution_cache_dir and repository_cache_dir to be the same.  

I'm assuming you're referring to the cache-dir option on the `ivy_subsystem` and not this one on the ivy_task_mixin?

Reply:
Thanks. I was kinda expecting some logic between old and new options. Maybe somewhere  around https://github.com/pantsbuild/pants/pull/5844/files/edcfcf1c2206aeab9e1398bd1b5a6252b9a00cbc#diff-23cdf27cceb5831fcb8c131fd37fe4cbR61 like
```
if ivy_cache_dir is not set:
  use the new options
```

Does that make sense?

Reply:
Are you looking for the logic in ivy_subsystem.py ?

```
+  def resolution_cache_dir(self):
+    if self.get_options().resolution_cache_dir:
+      return self.get_options().resolution_cache_dir
+    else:
+      return self.get_options().cache_dir
+
+  def repository_cache_dir(self):
+    if self.get_options().repository_cache_dir:
+      return self.get_options().repository_cache_dir
+    else:
+      return self.get_options().cache_dir
```

Reply:
@cheister ah yes, thanks!

Nit: 

1. Could you make the default value `None` explicitly for
```
+    register('--resolution-cache-dir', advanced=True,
+             help='Directory to store Ivy resolution artifacts.')
+    register('--repository-cache-dir', advanced=True,
+             help='Directory to store Ivy repository artifacts.')
```?

2. add some doc string for following 
```
+  def resolution_cache_dir(self):
+    if self.get_options().resolution_cache_dir:
+      return self.get_options().resolution_cache_dir
+    else:
+      return self.get_options().cache_dir
+
+  def repository_cache_dir(self):
+    if self.get_options().repository_cache_dir:
+      return self.get_options().repository_cache_dir
+    else:
+      return self.get_options().cache_dir
```
that the logic is for option deprecation handling?

Other than that lgtm!

Reply:
I originally had `None` for the default but was told it was redundant and removed it.

I also don't plan on deprecating `--cache-dir`, I think that will still be used most of the time when you don't want to share a cache between instances.   I updated the documentation for the option to more closely describe how it works with ivy itself.